pkg/rdt/rdt.go

```diff
@@ -80,54 +105,28 @@ func Initialize() error {
 	return nil
 }
 
-// GetClasses returns the names of the available RDT classes
-func GetClasses() []string {
-	return rdt.getClasses()
+// GetClass returns one RDT class
+func GetClass(name string) (CtrlGroup, bool) {
+	return rdt.getClass(name)
 }
 
-// SetProcessClass assigns a set of processes to a RDT class
-func SetProcessClass(class string, pids ...string) error {
-	return rdt.setProcessClass(class, pids...)
+// GetClasses returns all available RDT classes
+func GetClasses() map[string]CtrlGroup {
+	return rdt.getClasses()
 }
 
-func (c *control) getClasses() []string {
-	ret := make([]string, len(c.conf.Classes))
-
-	i := 0
-	for k := range c.conf.Classes {
-		ret[i] = k
-		i++
-	}
-	sort.Strings(ret)
-	return ret
+func (c *control) getClass(name string) (CtrlGroup, bool) {
+	cls, ok := c.classes[name]
+	return cls, ok
 }
 
-func (c *control) setProcessClass(class string, pids ...string) error {
-	if _, ok := c.conf.Classes[class]; !ok {
-		return rdtError("unknown RDT class %q", class)
-	}
+func (c *control) getClasses() map[string]CtrlGroup {
+	ret := make(map[string]CtrlGroup, len(c.classes))
 
-	path := filepath.Join(c.resctrlGroupPath(class), "tasks")
-	f, err := os.OpenFile(path, os.O_WRONLY, 0644)
-	if err != nil {
-		return err
-	}
-	defer f.Close()
-
-	for _, pid := range pids {
-		if _, err := f.WriteString(pid + "\n"); err != nil {
-			unwrapped := err
-			if pathError, ok := err.(*os.PathError); ok {
-				unwrapped = pathError.Unwrap()
-			}
-			if unwrapped == syscall.ESRCH {
-				c.Debug("no task %s", pid)
-			} else {
-				return rdtError("failed to assign processes %v to class %q: %v", pids, class, c.cmdError(err))
-			}
-		}
+	for k, v := range c.classes {
+		ret[k] = v
 	}
-	return nil
+	return ret
```

Comment:
Is it deliberate that, while we used to, we don't any more return classes in a stable (IOW sorted) order ?

Reply:
Hmm, good catch and good question. Yes, it was a deliberate change, but, in some ancient version of the patchset where it was convenient to have the quick lookup from map. However, that property is not used, anymore, and, I think it is probably better to stay in the sorted slice format. I'll change this accordingly

Reply:
Changed back to sorted slice.

I also change `CtrlGroup.GetMonGroups()` similarly